### PR TITLE
fix(opensearchtransport): default benchmark pprof to an ephemeral port

### PR DESCRIPTION
**Pull request by ryanyuan**
### Description

Back-to-back `go test -bench` runs could fail to start the pprof server because the previous run's socket was still in TIME_WAIT on a fixed port. This defaults pprof to an ephemeral loopback port (`localhost:0`), so repeated runs pick a free port instead of colliding.

The startup logic moves out of the benchmark test file into a small `internal/pprofutil` package. `PPROF_ADDR` sets the address: empty binds an ephemeral loopback port, and an explicit `host:port` (with or without an `http://` scheme) pins it, e.g. `localhost:6060`. Malformed input is logged and skipped rather than bound.

This is the last of the five benchmark bugs from #864; the other four were fixed in #859.

### Issues Resolved
Closes [#864]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: internal/pprofutil/pprofutil.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		return nil
+	}
+
+	ln, err := net.Listen("tcp", u.Host)
+	if err != nil {
+		log.Printf("pprof server failed to listen: %v", err)
+		return nil
+	}
+
+	log.Printf("pprof server listening on http://%s/debug/pprof/", ln.Addr())
```
Comment (sean-) [inline internal/pprofutil/pprofutil.go]: Why aren't we using `u.String()`?  The path` /debug/pprof` is also configurable, iirc, so we should use a local constant for this and also have a single-source-of-truth for the URL if not specified.

File: internal/pprofutil/pprofutil_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+		})
+	}
+}
+
+// TestStartNoCollisionOnRepeatedCalls guards issue #864 Bug 5. It proves
+// ephemeral-port allocation hands out distinct ports, so repeated startups do
+// not fight over one address. The real-world trigger -- a TIME_WAIT socket held
+// by a prior `go test -bench` process -- cannot be reproduced within a single
+// test binary, so two in-process starts is the closest observable proxy.
+func TestStartNoCollisionOnRepeatedCalls(t *testing.T) {
```
Comment (sean-) [inline internal/pprofutil/pprofutil_test.go]: Do we need to test the behavior of `:0`?

File: internal/pprofutil/pprofutil.go (hunk 1 above)
Comment (ryanyuan) [reply internal/pprofutil/pprofutil.go]: right now `/debug/pprof` is not configurable because the blank import (https://pkg.go.dev/net/http/pprof#section-documentation) hardcodes registration on DefaultServeMux in its init().

reference: https://pkg.go.dev/net/http/pprof#section-documentation

We could drop the blank import and register handlers by hand.

File: internal/pprofutil/pprofutil_test.go (hunk 2 above)
Comment (ryanyuan) [reply internal/pprofutil/pprofutil_test.go]: I think the `TestStart()` should cover the `:0` test case and make `:0` and `localhost:0` a duplicate here in this test case.

But we can also cover it here. WDYT?

Comment (sean-) [reply internal/pprofutil/pprofutil_test.go]: Please.  I want minimal redundancy and don't want a fluff test that tests Go's `:0` built-in.

File: internal/pprofutil/pprofutil.go (hunk 1 above)
Comment (sean-) [reply internal/pprofutil/pprofutil.go]: I think that's what's done elsewhere... or maybe not in this project, but yes.  For a generic helper func like this, that's how I'd spin things up (plus it also gets us a context on the http server which we can use to hook up signals in the future if we need to).

**Review by sean- — APPROVED** [review]
